src/software/logger/logger.h

```diff
@@ -9,17 +9,34 @@
 #include <g3log/logworker.hpp>
 
 #include "software/logger/coloured_cout_sink.h"
+#include "software/logger/csv_sink.h"
 #include "software/logger/custom_logging_levels.h"
 
-/**
- * This class acts as a Singleton that's responsible for initializing the logger.
- * We use a singleton rather than a generic function in this namespace because
- * the logWorker object must stay in scope, otherwise the logger is automatically
- * destroyed. So if an "init" function is used, the logWorker goes out of scope as
- * soon as the function returns, which destroys the logger right after creating it
- *
- * The Singleton class allows us to keep the logWorker in scope for the duration
- * of the program while still providing a single function to initialize the logger
+#define LOG_SELECT(_1, _2, NAME, ...) NAME
+
+#undef LOG
+#define LOG(...) LOG_SELECT(__VA_ARGS__, LOG_2, LOG_1)(__VA_ARGS__)
+#define LOG_1(level)                                                                     \
+    if (!g3::logLevel(level))                                                            \
+    {                                                                                    \
+    }                                                                                    \
+    else                                                                                 \
+        INTERNAL_LOG_MESSAGE(level).stream()
```

Comment:
can we put curly braces after the else for clarity?

Reply:
we end up with syntax errors because we need to pipe log messages